app/services/push_notification/account_delete.rb

```diff
@@ -0,0 +1,84 @@
+module PushNotification
+  class AccountDelete
+    def call(user_id)
```

Comment:
A suggestion for some simplifications here:

First, if you passed in the user instead of a user_id you could simplify some of these active record calls. For example, `AgencyIdentity.where(user_id: user_id)` could become `user.agency_identities`.

Secondly, if you saved the user in an initializer, it would save you from having to pass variables around as much. For example,

```
    def agency_id_to_uuid(user_id)
      hash = {}
      AgencyIdentity.where(user_id: user_id).each do |row|
        hash[row.agency_id] = row.uuid
      end
      hash
    end
```

...could become...

```
    def agency_uuid_for_sp(service_provider)
      @agency_id_to_agency_uuid_map = user.agency_identities.map do |row|
        [row.agency_id, row.uuid]
      end.to_h
      @agency_id_to_agency_uuid_map[service_provider.agency_id]
    end
```

And then you don't have to pass that map to every function that needs it. That let's you remove a parameter from several of your methods.

Reply:
I'm not in favor of passing a user an any of our service classes (vs a user_id).  It limits the flexibility of the interface.

Reply:
What do you mean?

Reply:
I meant if it needs a user and I don't have one I have to create one with the user_id just to satisfy the interface

Reply:
I went ahead and cleaned it up.  I originally broke it apart to satisfy the linter.  I reconsolidated it and made the linter happy.